src/gluonts/ext/hierarchicalforecast/_predictor.py

```diff
@@ -0,0 +1,309 @@
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.model.forecast import QuantileForecast
+from gluonts.ext.statsforecast import ModelConfig as _ModelConfig
+
+models_without_fitted_capability = [
+    "ADIDA",
+    "CrostonClassic",
+    "CrostonOptimized",
+    "CrostonSBA",
+    "IMAPA",
+    "SeasWA",
+    "TSB",
+    "WindowAverage",
+]
+
+
+def get_formatted_S(
+    _S: Union[List[List[int]], np.ndarray],
+    ts_names: List[Union[str, Any]],
+) -> pd.DataFrame:
+    S = np.array(_S)
+
+    return pd.DataFrame(
+        S, index=ts_names, columns=ts_names[S.shape[0] - S.shape[1] :]
+    )
+
+
+def format_data_entry(entry: DataEntry, S: pd.DataFrame) -> pd.DataFrame:
+    df = pd.DataFrame(entry["target"]).T
+    df.columns = S.index.tolist()
+    df.index = pd.date_range(
+        start=entry["start"].start_time,
+        periods=df.shape[0],
+        freq=entry["start"].freq,
+    )
+
+    df = unpivot(df)
+    df = df.set_index("unique_id")
+
+    return df
+
+
+def unpivot(frame: pd.DataFrame) -> pd.DataFrame:
+    n, k = frame.shape
+    return pd.DataFrame(
+        {
+            "unique_id": np.asarray(frame.columns).repeat(n),
+            "ds": np.tile(np.asarray(frame.index), k),
+            "y": frame.to_numpy().ravel("F"),
+        }
+    )
+
+
+def get_model_name(df: pd.DataFrame) -> Union[str, List[str]]:
+    df_columns = df.columns.to_list()
+    drop_cols = ["ds", "y"] if "y" in df_columns else ["ds"]
+    model_names = [x for x in df_columns if x not in drop_cols]
+
+    if len(model_names) == 1:
+        return model_names[0]
+
+    return model_names
+
+
+def format_reconciled_forecasts(
+    df: pd.DataFrame,
+    prediction_length: int,
+    fcst_col_name: str,
+    S: pd.DataFrame,
+):
+    target_dim = S.shape[0]
+
+    assert len(df) == target_dim * prediction_length
+
+    # we want rows to correspond to time and columns to forecasts
+    hier_forecasts = df.pivot(columns="ds", values=fcst_col_name)
+
+    # we want columns to follow the order of the rows of S
+    hier_forecasts = hier_forecasts.loc[S.index].T
+
+    # we want rows to be sorted according to time
+    hier_forecasts = hier_forecasts.sort_index()
+
+    return np.array(hier_forecasts)
+
+
+def keep_model_columns(df: pd.DataFrame, model_name: str) -> pd.DataFrame:
+    columns_to_keep = ["ds"]
+    columns_to_keep.extend([x for x in df.columns if model_name in x])
+
+    return df[columns_to_keep]
+
+
+def rename_model_columns(df: pd.DataFrame, model_name: str) -> pd.DataFrame:
+    mapper = {}
+    for e in df.columns:
+        if e == model_name:
+            mapper[e] = "mean"
+        else:
+            mapper[e] = e.replace(f"{model_name}-", "")
+
+    return df.rename(columns=mapper)
+
+
+def prune_fcst_df(
+    df: pd.DataFrame, base_reconciliation_model_name: str
+) -> pd.DataFrame:
+
+    df = keep_model_columns(df, base_reconciliation_model_name)
+    df = rename_model_columns(df, base_reconciliation_model_name)
```

Comment:
I think these two function are simple enough, that just having the entire behaviour here in `prune_fcst_df` would be easier to follow, maybe

Reply:
Yes, I updated the code accordingly.